tools/run_tests/jobset.py

```diff
@@ -167,25 +169,30 @@ class Job(object):
   def __init__(self, spec, bin_hash, newline_on_success, travis, add_env, xml_report):
     self._spec = spec
     self._bin_hash = bin_hash
+    self._newline_on_success = newline_on_success
+    self._travis = travis
+    self._add_env = add_env.copy()
+    self._xml_test = ET.SubElement(xml_report, 'testcase',
+                                   name=self._spec.shortname) if xml_report is not None else None
+    self._retries = 0
+    message('START', spec.shortname, do_newline=self._travis)
+    self.start()
+
+  def start(self):
     self._tempfile = tempfile.TemporaryFile()
     env = os.environ.copy()
-    for k, v in spec.environ.iteritems():
+    for k, v in self._spec.environ.iteritems():
```

Comment:
I realize that what caught my attention on these lines isn't what changed on these lines, but since it did catch my attention: why aren't these lines just
```
env = dict(os.environ)
env.update(self._spec.environ)
env.update(self._add_env)
```
?